mainapp/views.py

```diff
@@ -0,0 +1,202 @@
+from django.contrib import messages
+from django.contrib.auth.forms import AuthenticationForm
+from django.shortcuts import render, redirect, get_object_or_404
+from django.contrib.auth import login, authenticate
+from .form import Registr_new_user, ContactForm, Calc, CartaddForm, OrderCreateForm
+from django.core.mail import EmailMessage
+from django.contrib.auth.decorators import login_required
+from django.views.generic import DetailView, View
+from .models import Product, Categori, OrderItem
+from django.views.generic import ListView
+from .cart import Cart
+from django.views.decorators.http import require_POST
+from django.views.generic.edit import FormMixin
+
+
+class Productdetailvies(FormMixin, DetailView):
+    model = Product
+    template_name = "mainapp/Product_data.html"
+    context_object_name = 'Product'
+    form_class = CartaddForm
+    slug_url_kwarg = 'pcslug'
+
+    def get_context_data(self, **kwargs):
+        context = super(Productdetailvies, self).get_context_data(**kwargs)
+        context['form_class'] = self.get_form
+        return context
+
+
+class Startpage(View):
+
+    def get(self, request, *args, **kwargs):
+        categori = Categori.objects.all()
```

Comment:
Nie używasz kategorii w szablonie, więce pewnie można ją usunąć z kontekstu. Gdybyś chciał się odwoływać do kategorii produktu, to w django najprościej użyć metody select_related. Tzn. wyciągnąć wszystkie produkty, ale na querysecie wywołać funkcję select_related (ona wyciągnie z bazy i zcachuje powiązane obiekty).